scripts/downstream-tags.py

```
@@ -0,0 +1,191 @@
+    github_url = repo['github']
+    repo_name = github_url.replace('https://github.com/', '')
+
+    try:
+        result = subprocess.run(
+            ['gh', 'api', f'repos/{repo_name}/commits?path=lean-toolchain'],
+            capture_output=True,
+            text=True
+        )
+        if result.returncode != 0:
+            return None
+
+        commits = json.loads(result.stdout)
+
+        for commit in commits:
+            sha = commit['sha']
+            result = subprocess.run(
+                ['gh', 'api', f'repos/{repo_name}/contents/lean-toolchain?ref={sha}'],
+                capture_output=True,
+                text=True
+            )
+            if result.returncode == 0:
+                content = json.loads(result.stdout)
+                toolchain = base64.b64decode(content['content']).decode('utf-8').strip()
+                # Check for exact match
+                if toolchain == f'leanprover/lean4:{version}':
+                    return commit['sha']
+        return None
+    except Exception as e:
+        print(f"Error checking toolchain history for {repo['name']}: {e}", file=sys.stderr)
+        return None
+
+def get_current_toolchain(repo: Dict[str, str]) -> Optional[str]:
+    """Get the current toolchain version from the default branch"""
+    github_url = repo['github']
+    repo_name = github_url.replace('https://github.com/', '')
+
+    try:
+        result = subprocess.run(
+            ['gh', 'api', f'repos/{repo_name}/contents/lean-toolchain'],
+            capture_output=True,
+            text=True
+        )
+        if result.returncode == 0:
+            content = json.loads(result.stdout)
+            toolchain = base64.b64decode(content['content']).decode('utf-8').strip()
+            # Strip the prefix if present
+            return toolchain.replace('leanprover/lean4:', '')
+        return None
+    except Exception:
+        return None
+
+def main():
```

Could you please let this fail gracefully if `gh` is not installed?

---

Done.